src/main/python/pybuilder/plugins/python/coverage_plugin.py

```diff
@@ -316,7 +316,14 @@ def _delete_non_essential_modules():
     for module_name in list(sys.modules.keys()):
         module = sys.modules[module_name]
         if module:
-            if not _is_module_essential(module.__name__, sys_packages, sys_modules):
+            try:
+                # if we try to remove non-module object
+                module__name__ = module.__name__
```

Comment:
This is a bug in PyTest, that violates PEP-0320 and PEP-0420 since Python 3.3. Namely:

https://docs.python.org/3/reference/import.html#import-related-module-attributes

```
The import machinery fills in these attributes on each module object during loading, based on the module’s spec, before the loader executes the module.

__name__
The __name__ attribute **must be set to the fully-qualified name of the module**. This name is used to uniquely identify the module in the import system.
``` (emphasis added)

Please file a bug with PyTest.

Reply:
This PR is old as mammoth feces and Pytest doesn't use AliasModule anymore. I'll do changes next week and add `>=3.2.3` version of Pytest to plugin dependsies/

Reply:
Unfortunately, this problem was fixed, but still isn't released: https://github.com/pytest-dev/py/issues/73

Reply:
Asked about new release https://github.com/pytest-dev/py/issues/153

Reply:
So I'm not sure that `py` will be releases in the nearest future. I suggest to leave this changes and create new issue `revert coverage plugin changes related to py issue`. I'll periodically check `py` and do corresponding changes after `py` release.